Google map option added

ho-dor opened the pull request:
Fixes #1739

**Changes**: Option to use google map added along with OSM in Settings of app

**Screenshot/s for the changes**:
![20190605_234820](https://user-images.githubusercontent.com/37077735/58980244-7c110580-87ed-11e9-8f0c-1bcabae98baf.gif)


**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[g_map.zip](https://github.com/fossasia/pslab-android/files/3258527/g_map.zip)

ho-dor commented:
@neel1998 @CloudyPadmal please review

CloudyPadmal commented:
The moment we make a release with this inclusion, app will be rejected in FDroid. https://f-droid.org/en/docs/Inclusion_Policy/

We need a better solution..

iamareebjamal commented:
Create 2 build flavors, fDroid and playStore. See OpenEvent apps for reference

ho-dor commented:
> The moment we make a release with this inclusion, app will be rejected in FDroid. https://f-droid.org/en/docs/Inclusion_Policy/
> 
> We need a better solution..

How can we add google map option without the API key ? In #1739 it is mentioned that fdroid won't allow it and this option is for android only . So , how to build it only for android and not fdroid ?

iamareebjamal commented:
> So , how to build it only for android and not fdroid ?

I don't think you understand what f-droid is

ho-dor commented:
It's a catalogue of open source apps from what I understand.

On Thu, 6 Jun, 2019, 12:41 Areeb Jamal, <notifications@github.com> wrote:

> So , how to build it only for android and not fdroid ?
>
> I don't think you understand what f-droid is
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fossasia/pslab-android/pull/1775?email_source=notifications&email_token=AI24FZZV3AD6MMU4YPVPCI3PZC2CXA5CNFSM4HUC4TGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXB6HSY#issuecomment-499377099>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AI24FZYAQDBIYJMABIRGJJLPZC2CXANCNFSM4HUC4TGA>
> .
>

iamareebjamal commented:
So comparing android to f-droid is like comparing shampoo to lamborghini

ho-dor commented:
So our app is on fdroid because it has open source technologies and Google
map is not one so that's the problem. So , how to add it to our app ?

On Thu, 6 Jun, 2019, 12:44 Areeb Jamal, <notifications@github.com> wrote:

> So comparing android to f-droid is like comparing shampoo to lamborghini
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fossasia/pslab-android/pull/1775?email_source=notifications&email_token=AI24FZ6VW4L4XSLLHPOW62TPZC2N5A5CNFSM4HUC4TGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXB6NXA#issuecomment-499377884>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AI24FZ57KCL6ERYKWSFIZ4DPZC2N5ANCNFSM4HUC4TGA>
> .
>

neel1998 commented:
> So our app is on fdroid because it has open source technologies and Google map is not one so that's the problem. So , how to add it to our app ?
> […](#)
> On Thu, 6 Jun, 2019, 12:44 Areeb Jamal, ***@***.***> wrote: So comparing android to f-droid is like comparing shampoo to lamborghini — You are receiving this because you were assigned. Reply to this email directly, view it on GitHub <#1775?email_source=notifications&email_token=AI24FZ6VW4L4XSLLHPOW62TPZC2N5A5CNFSM4HUC4TGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXB6NXA#issuecomment-499377884>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AI24FZ57KCL6ERYKWSFIZ4DPZC2N5ANCNFSM4HUC4TGA> .

That's what we need to do. Have 2 different versions of app. One for play store with google maps and one for fdroid without google maps

iamareebjamal commented:
> So , how to add it to our app ?

Already told how https://github.com/fossasia/pslab-android/pull/1775#issuecomment-499373488

ho-dor commented:
> > So , how to add it to our app ?
> 
> Already told how [#1775 (comment)](https://github.com/fossasia/pslab-android/pull/1775#issuecomment-499373488)

Yes and I acknowledged it with a 👍

iamareebjamal commented:
Then what's the problem. You asked again how to do that. I supposed you got it

ho-dor commented:
No , actually your solution and my question was almost at the same time so , i did not see it then

CloudyPadmal commented:
@ho-dor you can get an idea from https://github.com/fossasia/open-event-android as @iamareebjamal mentioned..

ho-dor commented:
@CloudyPadmal https://www.androidauthority.com/building-multiple-flavors-android-app-706436/ is this the correct way to add a new flavour

iamareebjamal commented:
It's the same thing

mariobehling commented:
@ho-dor Could you please add the option directly on the settings screen instead of another popup? So, the user only has to click ones instead of twice?

ho-dor commented:
> @ho-dor Could you please add the option directly on the settings screen instead of another popup? So, the user only has to click ones instead of twice?

@mariobehling the current UI is more consistent with the app . For example , text format preference or the choosing the unit in thermometer settings or choosing the sensors in many instruments

ho-dor commented:
I need help in this , I am unable to understand the file structure that I have to create for flavours

neel1998 commented:
@ho-dor yes . I will start working on this

ho-dor commented:
@neel1998  I would actually like to learn the concept of build flavours so would be great if you could just guide me in the right direction so I can finalize this . Thanks

neel1998 commented:
@ho-dor sure. I will look into it how we can do it. And then will inform you so you can do it. Cheers:)

CloudyPadmal commented:
@ho-dor @neel1998 what are we gonna do with this? Any update?

neel1998 commented:
@ho-dor  you mentioned somewhere you have some idea to get this done?

ho-dor commented:
@neel1998  yes in the gitter channel but we were looking for better solutions as the mentioned solution would leave us with huge redundancy in code.

CloudyPadmal commented:
@ho-dor if we decouple Map rendering classes, that would not cause much redundancy right?

ho-dor commented:
@CloudyPadmal sir I have added the GoogleMapsActivity and the layout file activity_google_maps in playstore flavour and I have also added a settings_preference_xml in fdroid that does not have the maps preference option. Still a WIP. Getting errors like URI not registered in xml file for namespaces. Please guide further.

ho-dor commented:
If you will look at travis there is a strange error that I have not seen before . Merger error in manifest. Unable to solve it, have searched the net.

ho-dor commented:
Can you help me with this travis error, other than that fdroid and playstore variants look fine. @CloudyPadmal @neel1998

neel1998 commented:
@ho-dor  did you try this ?
``Suggestion: add 'tools:replace="android:appComponentFactory"' to <application> element at AndroidManifest.xml:22:5-129:19 to override.`` it is given in travis log

ho-dor commented:
Yes . But this attribute is already present in <application> with different
value

Regards,
Kunal Rai(BT16CSE058),
NATIONAL INSTITUTE OF TECHNOLOGY , UTTARAKHAND,
+91-7895780897

On Thu, 27 Jun, 2019, 12:26 Neel Trivedi, <notifications@github.com> wrote:

> @ho-dor <https://github.com/ho-dor> did you try this ?
> Suggestion: add 'tools:replace="android:appComponentFactory"' to
> <application> element at AndroidManifest.xml:22:5-129:19 to override. it
> is given in travis log
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fossasia/pslab-android/pull/1775?email_source=notifications&email_token=AI24FZ3RTKQ3KAHT2GGFDNTP4RQBNA5CNFSM4HUC4TGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODYWEM2Q#issuecomment-506218090>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AI24FZ5U2RS4UUKTDYZBITDP4RQBNANCNFSM4HUC4TGA>
> .
>

neel1998 commented:
@ho-dor  please check your PR. There are lot of code included from other PRs. Also somehow a lot of important code has been removed. Please check

ho-dor commented:
Yeah, I have messed up this branch pretty bad.

ho-dor commented:
@CloudyPadmal @neel1998 the problem is with SensorLoggerListAdapter class, it renders the map based on user preference . Now when I build the playstore flavour, no error since there is GoogleMapsActivity in that flavour but when I build fdroid flavour, it shows that it does not know any GoogleMapsActivity class so there is an error in SensorLoggerListAdapter class.

neel1998 commented:
> @CloudyPadmal @neel1998 the problem is with SensorLoggerListAdapter class, it renders the map based on user preference . Now when I build the playstore flavour, no error since there is GoogleMapsActivity in that flavour but when I build fdroid flavour, it shows that it does not know any GoogleMapsActivity class so there is an error in SensorLoggerListAdapter class.

Can't we put some conditions in the SensorListAdapter to check which map config is selected?

ho-dor commented on app/build.gradle:
```diff
@@ -31,6 +31,18 @@ android {
+        }
+        fdroid {
+            dimension = 'default'
+            applicationIdSuffix = '.fdroid'
```
I have followed SUSI.AI

ho-dor commented:
All these problems have occured due to the HEAD getting messed up for this branch , I will look into it.

ho-dor commented:
> > @CloudyPadmal @neel1998 the problem is with SensorLoggerListAdapter class, it renders the map based on user preference . Now when I build the playstore flavour, no error since there is GoogleMapsActivity in that flavour but when I build fdroid flavour, it shows that it does not know any GoogleMapsActivity class so there is an error in SensorLoggerListAdapter class.
> 
> Can't we put some conditions in the SensorListAdapter to check which map config is selected?

Even to put conditions, we will have to import GoogleMapsActivity atleast and that is the issue, for fdroid there is no GoogleMapsActivity
